dolphin/api/v1/storages.py

```diff
@@ -135,8 +134,35 @@ def update(self, req, id, body):
     def delete(self, req, id):
         return webob.Response(status_int=http_client.ACCEPTED)
 
+    @wsgi.response(202)
     def sync_all(self, req):
-        return dict(name="Sync all storages")
+        """
+        :param req:
+        :return: it's a Asynchronous call. so return 202 on success. sync_all
+        api performs the storage device info, pool, volume etc. tasks on each
+        registered storage device.
+        """
+        # validate the id
+        ctxt = req.environ['dolphin.context']
```

Comment:
Is this different from context.get_admin_context()?

Reply:
yes. an we should use, this in general users scenario.